pandas/core/strings.py

```diff
@@ -74,10 +74,20 @@ def cat_core(list_of_columns: List, sep: str):
     """
     if sep == "":
         # no need to interleave sep if it is empty
-        return np.sum(list_of_columns, axis=0)
+        with warnings.catch_warnings():
+            # See https://github.com/numpy/numpy/issues/15041
+            warnings.filterwarnings("ignore", ".*with automatic object dtype.*")
+            out = np.sum(list_of_columns, axis=0)
```

Comment:
can these be replaced with a ``sep.join(...)`` with a list-comprehension?

Reply:
maybe?  ATM im just trying to get a handle on the scope of the problem

Reply:
We will revert this for 1.18. PRobably also on master (but probably only temporarily to pacify downstream testsuits for a bit!). The interesting thing would be if there are cases where it cannot be fixed easily. I.e. adding a warning filter should not be necessary except for specific tests. (I am sure you are aware, but the filter context manager is not a good solution in this code, since it is not thread safe at all.)

Reply:
We will revert this for 1.18. PRobably also on master (but probably only temporarily to pacify downstream testsuits for a bit!). The interesting thing would be if there are cases where it cannot be fixed easily. I.e. adding a warning filter should not be necessary except for specific tests. (I am sure you are aware, but the filter context manager is not a good solution in this code, since it is not thread safe at all.)